platform-validation/scripts/analyzeSystemStats.sh

```diff
@@ -0,0 +1,245 @@
+               if [[ $overwrite -eq 1 ]]; then
+                    echo "$(date): Over-writing peak $type usage for $binary New: $value Old: $stored_peak_stat"
+                    sed -i "s/$binary-peak-$type=[0-9]*.[0-9]*/$binary-peak-$type=$value/" "$FILE"
+               fi
+          else
+               echo "$(date): Writing peak $type usage for $binary, Value: $value"
+               echo "$binary-peak-$type=$value" >>"$FILE"
+          fi
+
+          if [[ -n "$stored_avg_stat" ]]; then
+               avrgd_val=$(echo "$stored_avg_stat" "$value" | awk '{printf "%.2f", ($1+$2)/2}')
+               sed -i "s/$binary-avg-$type=[0-9]*.[0-9]*/$binary-avg-$type=$avrgd_val/" "$FILE"
+          else
+               echo "$(date): Writing avg $type usage for $binary, Value: $value"
+               echo "$binary-avg-$type=$value" >>"$FILE"
+          fi
+     fi
+
+}
+
+perform_analysis() {
+     IOTEDGE_BINARIES_SIZE=0.0
+     IOTEDGE_BINARIES_MEMORY=0.0
+     IOTEDGE_CONTAINERS_SIZE=0.0
+     IOTEDGE_CONTAINERS_MEMORY=0.0
+
+     for binary in $BINARIES; do
+          if [[ $binary =~ aziot-* ]]; then
+               binary_size="$(grep "$binary"-size <"$1" | sed -r "s/$binary-size=//g")"
+               memory_usage="$(grep "$binary"-avg-memory <"$1" | sed -r "s/$binary-avg-memory=//g")"
+               IOTEDGE_BINARIES_SIZE=$(echo "$IOTEDGE_BINARIES_SIZE" "$binary_size" | awk '{print $1 + $2}')
+               IOTEDGE_BINARIES_MEMORY=$(echo "$IOTEDGE_BINARIES_MEMORY" "$memory_usage" | awk '{print $1 + $2}')
+          fi
+     done
+
+     if [[ -n "$CONTAINERS" ]]; then
+
+          stored_total_container_size="$(grep "total-container-size" <"$1" | sed -r "s/total-container-size=//g")"
+          if [[ -n $stored_total_container_size ]]; then
+               for container in $CONTAINERS; do
+                    if [[ $container =~ edgeHub ]] || [[ $container =~ edgeAgent ]]; then
+                         memory_usage="$(grep "$container"-avg-memory <"$1" | sed -r "s/$container-avg-memory=//g")"
+                         IOTEDGE_CONTAINERS_MEMORY=$(echo "$IOTEDGE_CONTAINERS_MEMORY" "$memory_usage" | awk '{print $1 + $2}')
+                    else
+                         nonruntime_container_size="$(grep "$container"-size <"$1" | sed -r "s/$container-size=//g")"
+                         IOTEDGE_CONTAINERS_SIZE=$(echo "$stored_total_container_size" "$nonruntime_container_size" | awk '{print $1 - $2}')
+                    fi
+               done
+          else
+               echo "Total Container Size should be present, exiting"
+               exit 1
+          fi
+     fi
+
+     echo "iotedge-binaries-size=$IOTEDGE_BINARIES_SIZE"
+     echo "iotedge-binaries-avg-memory=$IOTEDGE_BINARIES_MEMORY"
+     echo "iotedge-container-size=$IOTEDGE_CONTAINERS_SIZE"
+     echo "iotedge-container-memory=$IOTEDGE_CONTAINERS_MEMORY"
+}
+
+process_args "$@"
+if [[ -n $MEMORY_FILE_PATH ]]; then
```

Comment:
What happens is `$MEMORY_FILE_PATH` isn't set? I ask because there's no `else` clause here.

Reply:
If the Path is not provided, the script simply starts gathering new statistics, creating a file. If it is provided, we simply perform some analysis on the file to get total iotedge binary size, total iotedge container size and total iotedge memory usage. I intend to use those numbers when doing storage and memory check in the Platform compatibility tool